### PR TITLE
docs: add a section to HACKING.md on PR titles (commit messages to main)

### PR DESCRIPTION
**Pull request by tonyandrewmeyer**
Explain that we use conventional commits, and list the types and scopes that we're starting with.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -160,6 +160,38 @@ your charm to a controller using that version of Juju. For example, with microk8
...
+Pull requests should have a short title that follows the
+[conventional commit style](https://www.conventionalcommits.org/en/) using one of these types:
+
+* chore
+* ci
+* docs
+* feat
+* fix
+* perf
+* refactor
```
Comment (benhoyt) [inline HACKING.md:176]: I think "refactor" is a "chore". Thoughts on getting rid of it? Or maybe we should go the other way -- what chores are there that *aren't* refactoring?

Also, what about dropping "perf"? I kinda think we'll almost never need this in Ops. But maybe it doesn't hurt, even as a signal that we are not unconcerned about perf.

Comment (tonyandrewmeyer) [reply HACKING.md:176]: For the record, I dropped "build" and "style" from the [Angular list that conventionalcommits.com recommends](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type).

If we get rid of both perf I would put a change that's purely performance improvements under refactor, so I'm not sure where that would go if we get rid of both. It doesn't seem like a chore, and maybe that is going too... I like keeping it, even if seldom-used, as a signal as you describe.

Other chores: dependency bumps, the release PRs. Looking back over the last few months, there are some that are "internal tooling" (they could probably have been categorised into one of the others). I'm +0 for keeping refactor and chore.

Comment (benhoyt) [reply HACKING.md:176]: Okay, sounds fine with me -- let's keep 'em. Being a little more specific doesn't hurt. As long as this list doesn't get too huge.

**Review by benhoyt — APPROVED** [review]

File: HACKING.md [hunk 2]
```diff
@@ -160,6 +160,38 @@ your charm to a controller using that version of Juju. For example, with microk8
...
+* revert
+* test
+
+If the PR is limited to one of the following scopes, also include the scope in the title,
+and otherwise leave off the scope.
+
+* harness
+
```
Comment (dimaqq) [inline HACKING.md:184]: Was there perhaps an intention to add more scope examples in this doc?
Otherwise "one of the following scopes" reads odd.

**Review by dimaqq — APPROVED** [review]

File: HACKING.md (hunk 2 above)
Comment (tonyandrewmeyer) [reply HACKING.md:184]: The intention (if I recall the discussion correctly) was that we would potentially add more scopes in the future, as we find something that would be useful. So I wrote it to easily accommodate that. I could rewrite to just focus specifically on `(harness)` for now if that would be better.

Comment (benhoyt) [reply HACKING.md:184]: Yeah, I wondered that too. I kind of agree with Dima -- would probably be better to have it read well for what it is now (just one scope) and update it when we add more. Or note explicitly that "at present there's only one" or something.

File: HACKING.md [hunk 3]
```diff
@@ -160,6 +160,36 @@ your charm to a controller using that version of Juju. For example, with microk8
...
+# Contributing
+
+Changes are proposed as [pull requests on GitHub](https://github.com/canonical/operator/pulls).
+
+Pull requests should have a short title that follows the
+[conventional commit style](https://www.conventionalcommits.org/en/) using one of these types:
+
+* chore
+* ci
+* docs
```
Comment (dimaqq) [inline HACKING.md:172]: 🤔 I tend to open PRs with `docs:` prefix and Tony with `doc:` prefix.

Should we settle on a single type or allow both?

Comment (benhoyt) [reply HACKING.md:172]: Yes, let's go with `docs`, as that's what https://www.conventionalcommits.org/en/ recommends.

**Comment by benhoyt** [comment]
@tonyandrewmeyer Is this good to merge?

File: HACKING.md (hunk 3 above)
Comment (tonyandrewmeyer) [reply HACKING.md:172]: Yeah, I've used `docs:` except I think just once, which was a typo on my part :disappointed:. This is where automation (in a follow-up PR, at some point in the future) would indeed be useful, to catch such errors.

**Comment by tonyandrewmeyer** [comment]
@IronCore864 are you ok with this as well? We've got two approvals (plus the implicit one from me as author) but it would be nice to have the whole team approve pre-merging since this is about team process rather than a specific feature.

**Review by IronCore864 — APPROVED** [review]

**Comment by IronCore864** [comment]
> @IronCore864 are you ok with this as well? We've got two approvals (plus the implicit one from me as author) but it would be nice to have the whole team approve pre-merging since this is about team process rather than a specific feature.

I'm completely OK with this. I like it that we keep "refactor", and only use scope for harness.

**Pull request merged**